tests/s25Main/simple/testMapBase.cpp

```diff
@@ -14,6 +14,43 @@
 
 BOOST_AUTO_TEST_SUITE(WorldCreationSuite)
 
+BOOST_AUTO_TEST_CASE(GetAllNeighboursUnion)
+{
+    MapBase world;
+
+    // No points -> empty response
+    std::vector<MapPoint> testPointsEmpty{};
+
+    auto replyEmpty = world.GetAllNeighboursUnion(testPointsEmpty);
+    BOOST_TEST_REQUIRE(replyEmpty.empty());
+
+    // Two compontents of 1 and 2 vertices
```

Comment:
Well, best explained here:
https://en.wikipedia.org/wiki/Component_(graph_theory)
https://en.wikipedia.org/wiki/Vertex_(graph_theory)

vertex is basically point
components are all connected points - two road networks in S2 are two components

Reply:
I meant what you refer with "vertex" here. Because the function takes as inputs a list of points. For the purpose of this function the points are not connected, i.e. there is no vertex between (10,10) and (10,11) here, they are just neighbours. Of course you could refer to the points as nodes in a graph which basically is the map and vertices would be the relationships, i.e. what makes points neighbours. So there is a vertex between (10,10) and (10,11)  but none of either to (1,1) (unless the map is small)

But I wouldn't get to technical in the comment here. So maybe:
```suggestion
    // 3 points where the last 2 share neighbours and are neighbours to each other
```

Because that is the important thing for the tested function: We get the neighbours but remove and duplicates and additionally also the points itself, but also avoid duplicates if one is the neighbour of another given point

Reply:
Yea, your comment works better.